src/contracts/GPv2Settlement.sol

```diff
@@ -154,15 +154,17 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         recoverOrderFromTrade(recoveredOrder, tokens, trade);
 
         IVault.SwapKind kind =
-            order.kind == GPv2Order.SELL
+            order.kind == GPv2Order.KIND_SELL
                 ? IVault.SwapKind.GIVEN_IN
                 : IVault.SwapKind.GIVEN_OUT;
 
         IVault.FundManagement memory funds;
         funds.sender = recoveredOrder.owner;
-        funds.fromInternalBalance = order.useInternalSellTokenBalance;
+        funds.fromInternalBalance =
+            order.sellTokenBalance == GPv2Order.BALANCE_INTERNAL;
```

Comment:
I find it slightly weird that we can settle `GPv2Order.BALANCE_ERC20` orders here, assuming the user has authorized the balancer vault. Maybe we should respect the signed order balance source and revert on an ERC20 value? 

Alternatively we could make our own AllowanceManager Balancer fast path aware (although I would assume users that want to benefit from the fast path are also happy to approve the vault directly).

Reply:
This PR just adds the options but doesn't do anything with them:

> Note, that currently the SC treat ERC20 and EXTERNAL the same for now, the actual differentiation will come in a follow up PR.

Reply:
Also, if we implement it as https://github.com/gnosis/gp-v2-contracts/issues/534 we would be withdrawing to the settlement contract first anyway, so we should be able to support `GPv2Order.BALANCE_ERC20` as well.